assignments/models.py

```diff
@@ -45,20 +49,73 @@ class Meta:
         verbose_name_plural = _('sections')
 
     def __str__(self):
-        return 'Section: {}'.format(self.title)
+        return self.title
+
+    def get_absolute_url(self):
+        return reverse('assignments:section-detail', args=[self.assignment.slug, self.pk])
 
 
-class OpenTextTask(models.Model):
+class BaseTask(models.Model):
+    """
+    Abstract model for all tasks.
+    """
+    section = models.ForeignKey(Section, related_name='%(class)ss')
+
+    class Meta:
+        abstract = True
+
+
+class OpenTextTask(BaseTask):
     """
     Simple question task where the answer is placed in text area
     """
-    section = models.ForeignKey(Section, related_name='open_text_tasks')
     question = models.TextField()
 
     class Meta:
         verbose_name = _('open text task')
         verbose_name_plural = _('open text tasks')
 
+
+class BudgetingTarget(models.Model):
+    """
+    Budget task targets. Not all the fields have to be defined here. Reference amount refers to official amount spent
+    on this target. Lower and upper bounds should be defined only for targets that value can be in certain range.
+    Icon should be defined for targets that can be placed on the map.
+    """
+
+    name = models.CharField(_('name'), max_length=256)
+    unit_price = models.DecimalField(_('price'), max_digits=10, decimal_places=2, default=0)
+    reference_amount = models.DecimalField(_('reference amount'), max_digits=10, decimal_places=2, default=0)
+    lower_bound = models.DecimalField(_('lower bound'), max_digits=10, decimal_places=2, null=True, blank=True)
+    upper_bound = models.DecimalField(_('upper bound'), max_digits=10, decimal_places=2, null=True, blank=True)
+    icon = models.FileField(upload_to='map/icons/', null=True, blank=True)
+
+    class Meta:
+        verbose_name = _('budget target')
+        verbose_name_plural = _('budget targets')
+
     def __str__(self):
-        return 'Question: {}'.format(self.question)
+        return self.name
+
 
+class BudgetingTask(BaseTask):
```

Comment:
Budgeting task can have also 'allowed amount' which can use for selected budgeting targets. See the excel file (käyttämättä).